applications/ColossalQA/examples/retrieval_conversation_en.py

```diff
@@ -0,0 +1,89 @@
+'''
+Script for English retrieval based conversation system backed by LLaMa2
+'''
+import os
+from langchain.embeddings import HuggingFaceEmbeddings
+from langchain import LLMChain
+from colossalqa.chain.retrieval_qa.base import RetrievalQA
+from langchain.vectorstores import Chroma
+from colossalqa.local.llm import VllmLLM, ColossalAPI, ColossalLLM
+from colossalqa.data_loader.document_loader import DocumentLoader
+from colossalqa.retriever import CustomRetriever
+from colossalqa.text_splitter import NeuralTextSplitter
+from colossalqa.memory import ConversationBufferWithSummary
+from colossalqa.prompt.prompt import PROMPT_RETRIEVAL_QA_EN, PROMPT_DISAMBIGUATE_EN
+
+# vllm
+# start the vllm server with
+# python -m vllm.entrypoints.api_server --model "/path to model/Llama-2-7b-hf" --swap-space 16 --disable-log-requests --host localhost --port 8077 --max-num-seqs 256 --gpu-memory-utilization 0.5
+
+# setup LLM
+# llm = VllmLLM(n=1)
+
+# local coati api
+
+model_path = os.environ.get('EN_MODEL_PATH')
+model_name = os.environ.get('EN_MODEL_NAME')
+colossal_api = ColossalAPI(model_name, model_path)
+llm = ColossalLLM(n=1, api=colossal_api)
+
+# define the retriever
+information_retriever = CustomRetriever(k=3)
+
+# setup embedding model locally
+embedding = HuggingFaceEmbeddings(model_name="moka-ai/m3e-base",
+                           model_kwargs={'device': 'cpu'},encode_kwargs={'normalize_embeddings': False})
+
+# define memory with summarization ability
+memory = ConversationBufferWithSummary(llm=llm, max_tokens=2000,
+        llm_kwargs={'max_new_tokens':50, 'temperature':0.6, 'do_sample':True})
+
+# define the chain to preprocess the input
+# disambiguate the input. e.g. "What is the capital of that country?" -> "What is the capital of France?"
+llm_chain_disambiguate = LLMChain(llm=llm, prompt=PROMPT_DISAMBIGUATE_EN, llm_kwargs={'max_new_tokens':30, 'temperature':0.6, 'do_sample':True})
+
+def disambiguity(input):
+    out = llm_chain_disambiguate.run(input=input, chat_history=memory.buffer, stop=['\n'])
+    return out.split('\n')[0] 
+
+if __name__ == '__main__':
+    # Load data to vector store
+    print("Select files for constructing retriever")
+    documents = []
+    while True:
+        file = input("Select a file to load or enter Esc to exit:")
+        if file=='Esc':
+            break
+        data_name = input("Enter a short description of the data:")
+        retriever_data = DocumentLoader([[file, data_name.replace(' ', '_')]]).all_data
```

Comment:
For these examples, the steps should be the same and format should be the same.

For example, why this file has  "if __name__ == '__main__'" while the others are not?